src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -1270,6 +1270,32 @@ public Page search(String q) throws Exception {
      * @return The JSON.
      */
     public JSONWebResponse getJSON(@NonNull String path) throws IOException {
+
+        JenkinsRule.WebClient webClient = createWebClient();
+        return getJSON(path, webClient);
+    }
+
+    public JSONWebResponse getJSON(@NonNull String path, @NonNull JenkinsRule.WebClient webClient) throws IOException {
+        assert !path.startsWith("/");
+
+        Page runsPage = null;
+        try {
+            runsPage = webClient.goTo(path, "application/json");
+        } catch (SAXException e) {
+            // goTo shouldn't be throwing a SAXException for JSON.
+            throw new IllegalStateException("Unexpected SAXException.", e);
+        }
+        WebResponse webResponse = runsPage.getWebResponse();
+
+        return new JSONWebResponse(webResponse);
+    }
+
+    /**
+     * Put JSON from A Jenkins endpoint.
+     * @param path The endpoint URL.
+     * @return The JSON.
+     */
+    public JSONWebResponse putJSON(@NonNull String path) throws IOException {
```

Comment:
How is this “put”? Still looks like a `GET` to me.

Reply:
I am not done coding at all, I did a draft to find why I cannot test it first with `simpleGetShouldWork`